libs/features/placeholders.js

```diff
@@ -1,20 +1,20 @@
+import { customFetch, getConfig } from '../utils/utils.js';
+
 const fetchedPlaceholders = {};
 
 const getPlaceholdersPath = (config, sheet) => {
   const path = `${config.locale.contentRoot}/placeholders.json`;
   const query = sheet !== 'default' && typeof sheet === 'string' && sheet.length ? `?sheet=${sheet}` : '';
   return `${path}${query}`;
 };
-
-const fetchPlaceholders = async (config, sheet) => {
-  const placeholdersPath = getPlaceholdersPath(config, sheet);
-  const { customFetch } = await import('../utils/helpers.js');
-
-  fetchedPlaceholders[placeholdersPath] = fetchedPlaceholders[placeholdersPath]
+const fetchPlaceholders = async ({ config, sheet, placeholderRequest, placeholderPath }) => {
+  const path = placeholderPath || getPlaceholdersPath(config, sheet);
+  fetchedPlaceholders[path] = fetchedPlaceholders[path]
     // eslint-disable-next-line no-async-promise-executor
     || new Promise(async (resolve) => {
-      const resp = await customFetch({ resource: placeholdersPath, withCacheRules: true })
-        .catch(() => ({}));
+      const resp = await placeholderRequest
+        || await customFetch({ resource: path, withCacheRules: true })
+          .catch(() => ({}));
```

Comment:
Nit: we can remove some parentheses from here 
```suggestion
        || await customFetch({ resource: path, withCacheRules: true })
          .catch(() => {});
```

Reply:
`({})` returns an empty object vs `{}` would implicity return undefined.

As we check `const json = resp.ok ? await resp.json() : { data: [] };` within the placeholders, removing the parentheses would lead to a JS error